Distance conversion formulas and tests

eternal44 opened the pull request:
All formula methods are tested and pass.

@kotp - I played around with our new api contract and I think I get it.  
- The proc and the lambda we're using for the conversion formula methods are like 'a += b' where we're modifying 'a' with our proc calls.  
- we should change the way we wrap our class in a module.  If we're not using the module for namespacing we could remove the 'Convertor' module and our gem should still work.

eternal44 committed:
Remove redundancy from conversion file names

Ex: distance_conversion is repetitive in regards to it's parent
directory 'unit_conversion'.

eternal44 committed:
Full list of distance conversion formulas & tests

All distance methods pass tests.

kotp commented:
Looks good.  I am not sure about the 'setup' method, exactly.  

`Distance.new(1000, 'miles')` should be the expectation for the user.  Once that object is created, then we should be able to simply call .to_inches on it.

The `setup` method can still exist, obviously, but I think it should be internal use only.

eternal44 committed:
Replace setup method with class initialize method

Allows for simpler api contract

eternal44 committed:
Progress report for test results

Because pretty test results never hurt anybody :)

eternal44 commented:
Re: 'setup' method - yeah I forget why we decided to do it that way but I fixed it to simplify our api contract the way you described.  

Also added some color to our reports.  If all checks out let me know and I'll merge this PR in.
